### ISSUE TITLE
Error in check_sen2r_deps(): wmic not found

### ISSUE THREAD

**Issue by Pakillo** [ranghetti/sen2r#227]
Hello,

First, thanks for the package. Looks great!

I've just installed it from CRAN and run it for the first time, but running `check_sen2r_deps()` as recommended is throwing an error (in Windows):

> check_sen2r_deps()
Error in system("wmic logicaldisk get Caption", intern = T) : 
  'wmic' not found

I fixed it by adding Wbem to the Windows path:

```r
old_path <- Sys.getenv("PATH")
new_path <- "C:/windows/System32/Wbem"
Sys.setenv(PATH = paste(old_path, new_path, sep = ";"))
```

I think this error may disconcert new users. Maybe the function could warn if wmic is not found in the path, or explain in the documentation that wmic should be added to the path.

Reproducible example below.

Thank you!

``` r
library(sen2r)
#> Welcome to sen2r. To use the package from a GUI, launch
#>  > sen2r()
#> Documentation: http://sen2r.ranghetti.info
#> 
#> IMPORTANT: sen2r depends on some external tools;
#> before using it, it is strongly recommended to run function
#>  > check_sen2r_deps()
#> to check them and install the missing ones.
check_sen2r_deps()
#> Error in system("wmic logicaldisk get Caption", intern = T): 'wmic' not found
```

<sup>Created on 2019-10-23 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> - Session info ----------------------------------------------------------
#>  setting  value                       
#>  version  R version 3.6.1 (2019-07-05)
#>  os       Windows 7 x64 SP 1          
#>  system   x86_64, mingw32             
#>  ui       RTerm                       
#>  language (EN)                        
#>  collate  English_United Kingdom.1252 
#>  ctype    English_United Kingdom.1252 
#>  tz       Europe/Paris                
#>  date     2019-10-23                  
#> 
#> - Packages --------------------------------------------------------------
#>  package        * version   date       lib source        
#>  abind            1.4-5     2016-07-21 [1] CRAN (R 3.6.0)
#>  assertthat       0.2.1     2019-03-21 [1] CRAN (R 3.6.1)
#>  backports        1.1.5     2019-10-02 [1] CRAN (R 3.6.1)
#>  callr            3.3.2     2019-09-22 [1] CRAN (R 3.6.1)
#>  class            7.3-15    2019-01-01 [2] CRAN (R 3.6.1)
#>  classInt         0.4-2     2019-10-17 [1] CRAN (R 3.6.1)
#>  cli              1.1.0     2019-03-19 [1] CRAN (R 3.6.1)
#>  codetools        0.2-16    2018-12-24 [2] CRAN (R 3.6.1)
#>  colorspace       1.4-1     2019-03-18 [1] CRAN (R 3.6.1)
#>  crayon           1.3.4     2017-09-16 [1] CRAN (R 3.6.1)
#>  crosstalk        1.0.0     2016-12-21 [1] CRAN (R 3.6.1)
#>  curl             4.2       2019-09-24 [1] CRAN (R 3.6.1)
#>  data.table       1.12.4    2019-10-03 [1] CRAN (R 3.6.1)
#>  DBI              1.0.0     2018-05-02 [1] CRAN (R 3.6.1)
#>  desc             1.2.0     2018-05-01 [1] CRAN (R 3.6.1)
#>  devtools         2.2.1     2019-09-24 [1] CRAN (R 3.6.1)
#>  digest           0.6.21    2019-09-20 [1] CRAN (R 3.6.1)
#>  doParallel       1.0.15    2019-08-02 [1] CRAN (R 3.6.1)
#>  e1071            1.7-2     2019-06-05 [1] CRAN (R 3.6.1)
#>  ellipsis         0.3.0     2019-09-20 [1] CRAN (R 3.6.1)
#>  evaluate         0.14      2019-05-28 [1] CRAN (R 3.6.1)
#>  fastmap          1.0.1     2019-10-08 [1] CRAN (R 3.6.1)
#>  foreach          1.4.7     2019-07-27 [1] CRAN (R 3.6.1)
#>  foreign          0.8-72    2019-08-02 [2] CRAN (R 3.6.1)
#>  fs               1.3.1     2019-05-06 [1] CRAN (R 3.6.1)
#>  geojson          0.3.2     2019-01-31 [1] CRAN (R 3.6.1)
#>  geojsonio        0.7.0     2019-04-25 [1] CRAN (R 3.6.1)
#>  glue             1.3.1     2019-03-12 [1] CRAN (R 3.6.1)
#>  highr            0.8       2019-03-20 [1] CRAN (R 3.6.1)
#>  htmltools        0.4.0     2019-10-04 [1] CRAN (R 3.6.1)
#>  htmlwidgets      1.5.1     2019-10-08 [1] CRAN (R 3.6.1)
#>  httpuv           1.5.2     2019-09-11 [1] CRAN (R 3.6.1)
#>  httr             1.4.1     2019-08-05 [1] CRAN (R 3.6.1)
#>  iterators        1.0.12    2019-07-26 [1] CRAN (R 3.6.1)
#>  jqr              1.1.0     2018-10-22 [1] CRAN (R 3.6.1)
#>  jsonlite         1.6       2018-12-07 [1] CRAN (R 3.6.1)
#>  KernSmooth       2.23-16   2019-10-15 [2] CRAN (R 3.6.1)
#>  knitr            1.25      2019-09-18 [1] CRAN (R 3.6.1)
#>  later            1.0.0     2019-10-04 [1] CRAN (R 3.6.1)
#>  lattice          0.20-38   2018-11-04 [2] CRAN (R 3.6.1)
#>  lazyeval         0.2.2     2019-03-15 [1] CRAN (R 3.6.1)
#>  leaflet          2.0.2     2018-08-27 [1] CRAN (R 3.6.1)
#>  leaflet.extras   1.0.0     2018-04-21 [1] CRAN (R 3.6.1)
#>  lwgeom           0.1-7     2019-05-06 [1] CRAN (R 3.6.1)
#>  magrittr         1.5       2014-11-22 [1] CRAN (R 3.6.1)
#>  mapedit          0.5.0     2019-03-16 [1] CRAN (R 3.6.1)
#>  maptools         0.9-8     2019-10-05 [1] CRAN (R 3.6.1)
#>  Matrix           1.2-17    2019-03-22 [2] CRAN (R 3.6.1)
#>  memoise          1.1.0     2017-04-21 [1] CRAN (R 3.6.1)
#>  mime             0.7       2019-06-11 [1] CRAN (R 3.6.0)
#>  munsell          0.5.0     2018-06-12 [1] CRAN (R 3.6.1)
#>  pillar           1.4.2     2019-06-29 [1] CRAN (R 3.6.1)
#>  pkgbuild         1.0.6     2019-10-09 [1] CRAN (R 3.6.1)
#>  pkgconfig        2.0.3     2019-09-22 [1] CRAN (R 3.6.1)
#>  pkgload          1.0.2     2018-10-29 [1] CRAN (R 3.6.1)
#>  prettyunits      1.0.2     2015-07-13 [1] CRAN (R 3.6.1)
#>  processx         3.4.1     2019-07-18 [1] CRAN (R 3.6.1)
#>  promises         1.1.0     2019-10-04 [1] CRAN (R 3.6.1)
#>  ps               1.3.0     2018-12-21 [1] CRAN (R 3.6.1)
#>  R6               2.4.0     2019-02-14 [1] CRAN (R 3.6.1)
#>  raster           3.0-7     2019-09-24 [1] CRAN (R 3.6.1)
#>  Rcpp             1.0.2     2019-07-25 [1] CRAN (R 3.6.1)
#>  remotes          2.1.0     2019-06-24 [1] CRAN (R 3.6.1)
#>  reticulate       1.13      2019-07-24 [1] CRAN (R 3.6.1)
#>  rgdal            1.4-6     2019-10-01 [1] CRAN (R 3.6.1)
#>  rgeos            0.5-2     2019-10-03 [1] CRAN (R 3.6.1)
#>  rlang            0.4.0     2019-06-25 [1] CRAN (R 3.6.1)
#>  rmarkdown        1.16      2019-10-01 [1] CRAN (R 3.6.1)
#>  rprojroot        1.3-2     2018-01-03 [1] CRAN (R 3.6.1)
#>  scales           1.0.0     2018-08-09 [1] CRAN (R 3.6.1)
#>  sen2r          * 1.1.0     2019-10-21 [1] CRAN (R 3.6.1)
#>  sessioninfo      1.1.1     2018-11-05 [1] CRAN (R 3.6.1)
#>  sf               0.8-0     2019-09-17 [1] CRAN (R 3.6.1)
#>  shiny            1.4.0     2019-10-10 [1] CRAN (R 3.6.1)
#>  shinydashboard   0.7.1     2018-10-17 [1] CRAN (R 3.6.1)
#>  shinyFiles       0.7.3     2019-04-30 [1] CRAN (R 3.6.1)
#>  shinyjs          1.0       2018-01-08 [1] CRAN (R 3.6.1)
#>  shinyWidgets     0.4.9     2019-09-10 [1] CRAN (R 3.6.1)
#>  sp               1.3-1     2018-06-05 [1] CRAN (R 3.6.1)
#>  stars            0.4-0     2019-10-10 [1] CRAN (R 3.6.1)
#>  stringi          1.4.3     2019-03-12 [1] CRAN (R 3.6.0)
#>  stringr          1.4.0     2019-02-10 [1] CRAN (R 3.6.1)
#>  testthat         2.2.1     2019-07-25 [1] CRAN (R 3.6.1)
#>  tibble           2.1.3     2019-06-06 [1] CRAN (R 3.6.1)
#>  units            0.6-5     2019-10-08 [1] CRAN (R 3.6.1)
#>  usethis          1.5.1     2019-07-04 [1] CRAN (R 3.6.1)
#>  V8               2.3       2019-07-02 [1] CRAN (R 3.6.1)
#>  withr            2.1.2     2018-03-15 [1] CRAN (R 3.6.1)
#>  xfun             0.10      2019-10-01 [1] CRAN (R 3.6.1)
#>  XML              3.98-1.20 2019-06-06 [1] CRAN (R 3.6.0)
#>  xtable           1.8-4     2019-04-21 [1] CRAN (R 3.6.1)
#>  yaml             2.2.0     2018-07-25 [1] CRAN (R 3.6.0)
#> 
#> [1] C:/Users/FRS/Documents/R/win-library/3.6
#> [2] C:/R/R-3.6.1/library
```

</details>

**Comment by lbusett** [comment]
@Pakillo 

 First time we see this: thanks for signaling it.

 Digging a bit, it seems an "erratic" problem related to `ShinyFiles`: 

https://github.com/thomasp85/shinyFiles/issues/85 
https://github.com/thomasp85/shinyFiles/pull/41

@thomasp85 @vnijs do you know of any alternatives to the "Sys.setenv" solution?

**Comment by vnijs** [comment]
Thanks for reporting @lbusett.  Can you open a new issue at https://github.com/thomasp85/shinyFiles/issues? Thanks

FYI I cannot change the users path in shinyFiles but can at least test of wmic is available in the path and fail a bit more elegantly with a message to the user. I am not an expert on Windows systems so I you or @Pakillo want to file a PR for shinyFiles with an alternative to wmic (if that exists) I'd be happy to take a look at it

**Comment by vnijs** [comment]
FYI I pushed an update that will at least avoid the Error. There is no way an R package would be allowed on CRAN that would adjust the users PATH variable. If there is some other/better command line tool on windows that can provide information on drives/volumes I'd be happy to review a PR. You can see the changes at the link below. Note that "where wmicx" was a deliberate typo I had used in testing. I fixed that in the next commit to "where wmic"

https://github.com/thomasp85/shinyFiles/commit/762fb73e7a0226e03fc29501b865e0be89da19c9

To try out this "fix", please in stall from github using the command below.

```
remotes::install_github("thomasp85/shinyFiles")
```

**Comment by lbusett** [comment]
@vnijs thanks for looking in to this! I'll see if I can find a different solution for getting volumes in Windows

@Pakillo could you try installing shinyFiles from github as suggested and see what happens if you try again running `sen2r()`?

Lorenzo

**Comment by lbusett** [comment]
@Pakillo 

could you try running this command in a `cmd` terminal and see if it works? 

` %SystemRoot%\System32\Wbem\wmic logicaldisk get Caption`

**Comment by Pakillo** [comment]
Thank you all for the quick replies and fixes!

I've installed the last version of shinyFiles from GitHub, and run `check_sen2r_deps()` again. The shiny app works nice (because I already added Wbem to Windows path yesterday), but now I'm getting this new warning in the console (full reprex below):
```
The wmic command does not seem to be in your system path. For a possible
fix, see the link below or contact your IT department for help.
https://github.com/thomasp85/shinyFiles/issues/85#issuecomment-494601646

Listening on http://127.0.0.1:6406
Warning message:
In system("where wmic", intern = T) :
  running command 'where wmic' had status 1
```

This seems strange since wmic/Wbem is already in the path and the shiny app is working nicely?

---

@lbusett I run `%SystemRoot%\System32\Wbem\wmic logicaldisk get Caption` in the cmd terminal and it works (i.e. outputs drives names)

Thanks


``` r
library(sen2r)
#> Welcome to sen2r. To use the package from a GUI, launch
#>  > sen2r()
#> Documentation: http://sen2r.ranghetti.info
#> 
#> IMPORTANT: sen2r depends on some external tools;
#> before using it, it is strongly recommended to run function
#>  > check_sen2r_deps()
#> to check them and install the missing ones.
check_sen2r_deps()
#> Warning in system("where wmic", intern = T): running command 'where wmic'
#> had status 1
#> The wmic command does not seem to be in your system path. For a possible
#> fix, see the link below or contact your IT department for help.
#> https://github.com/thomasp85/shinyFiles/issues/85#issuecomment-494601646
#> Error in check_sen2r_deps(): The function must be run from an interactive R session.
```

<sup>Created on 2019-10-24 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> - Session info ----------------------------------------------------------
#>  setting  value                       
#>  version  R version 3.6.1 (2019-07-05)
#>  os       Windows 7 x64 SP 1          
#>  system   i386, mingw32               
#>  ui       RTerm                       
#>  language (EN)                        
#>  collate  English_United Kingdom.1252 
#>  ctype    English_United Kingdom.1252 
#>  tz       Europe/Paris                
#>  date     2019-10-24                  
#> 
#> - Packages --------------------------------------------------------------
#>  package        * version   date       lib
#>  abind            1.4-5     2016-07-21 [1]
#>  assertthat       0.2.1     2019-03-21 [1]
#>  backports        1.1.5     2019-10-02 [1]
#>  callr            3.3.2     2019-09-22 [1]
#>  class            7.3-15    2019-01-01 [2]
#>  classInt         0.4-2     2019-10-17 [1]
#>  cli              1.1.0     2019-03-19 [1]
#>  codetools        0.2-16    2018-12-24 [2]
#>  colorspace       1.4-1     2019-03-18 [1]
#>  crayon           1.3.4     2017-09-16 [1]
#>  crosstalk        1.0.0     2016-12-21 [1]
#>  curl             4.2       2019-09-24 [1]
#>  data.table       1.12.4    2019-10-03 [1]
#>  DBI              1.0.0     2018-05-02 [1]
#>  desc             1.2.0     2018-05-01 [1]
#>  devtools         2.2.1     2019-09-24 [1]
#>  digest           0.6.21    2019-09-20 [1]
#>  doParallel       1.0.15    2019-08-02 [1]
#>  e1071            1.7-2     2019-06-05 [1]
#>  ellipsis         0.3.0     2019-09-20 [1]
#>  evaluate         0.14      2019-05-28 [1]
#>  fastmap          1.0.1     2019-10-08 [1]
#>  foreach          1.4.7     2019-07-27 [1]
#>  foreign          0.8-72    2019-08-02 [2]
#>  fs               1.3.1     2019-05-06 [1]
#>  geojson          0.3.2     2019-01-31 [1]
#>  geojsonio        0.7.0     2019-04-25 [1]
#>  glue             1.3.1     2019-03-12 [1]
#>  highr            0.8       2019-03-20 [1]
#>  htmltools        0.4.0     2019-10-04 [1]
#>  htmlwidgets      1.5.1     2019-10-08 [1]
#>  httpuv           1.5.2     2019-09-11 [1]
#>  httr             1.4.1     2019-08-05 [1]
#>  iterators        1.0.12    2019-07-26 [1]
#>  jqr              1.1.0     2018-10-22 [1]
#>  jsonlite         1.6       2018-12-07 [1]
#>  KernSmooth       2.23-16   2019-10-15 [2]
#>  knitr            1.25      2019-09-18 [1]
#>  later            1.0.0     2019-10-04 [1]
#>  lattice          0.20-38   2018-11-04 [2]
#>  lazyeval         0.2.2     2019-03-15 [1]
#>  leaflet          2.0.2     2018-08-27 [1]
#>  leaflet.extras   1.0.0     2018-04-21 [1]
#>  lwgeom           0.1-7     2019-05-06 [1]
#>  magrittr         1.5       2014-11-22 [1]
#>  mapedit          0.5.0     2019-03-16 [1]
#>  maptools         0.9-8     2019-10-05 [1]
#>  Matrix           1.2-17    2019-03-22 [2]
#>  memoise          1.1.0     2017-04-21 [1]
#>  mime             0.7       2019-06-11 [1]
#>  munsell          0.5.0     2018-06-12 [1]
#>  pillar           1.4.2     2019-06-29 [1]
#>  pkgbuild         1.0.6     2019-10-09 [1]
#>  pkgconfig        2.0.3     2019-09-22 [1]
#>  pkgload          1.0.2     2018-10-29 [1]
#>  prettyunits      1.0.2     2015-07-13 [1]
#>  processx         3.4.1     2019-07-18 [1]
#>  promises         1.1.0     2019-10-04 [1]
#>  ps               1.3.0     2018-12-21 [1]
#>  R6               2.4.0     2019-02-14 [1]
#>  raster           3.0-7     2019-09-24 [1]
#>  Rcpp             1.0.2     2019-07-25 [1]
#>  remotes          2.1.0     2019-06-24 [1]
#>  reticulate       1.13      2019-07-24 [1]
#>  rgdal            1.4-6     2019-10-01 [1]
#>  rgeos            0.5-2     2019-10-03 [1]
#>  rlang            0.4.0     2019-06-25 [1]
#>  rmarkdown        1.16      2019-10-01 [1]
#>  rprojroot        1.3-2     2018-01-03 [1]
#>  scales           1.0.0     2018-08-09 [1]
#>  sen2r          * 1.1.0     2019-10-21 [1]
#>  sessioninfo      1.1.1     2018-11-05 [1]
#>  sf               0.8-0     2019-09-17 [1]
#>  shiny            1.4.0     2019-10-10 [1]
#>  shinydashboard   0.7.1     2018-10-17 [1]
#>  shinyFiles       0.7.5     2019-10-24 [1]
#>  shinyjs          1.0       2018-01-08 [1]
#>  shinyWidgets     0.4.9     2019-09-10 [1]
#>  sp               1.3-1     2018-06-05 [1]
#>  stars            0.4-0     2019-10-10 [1]
#>  stringi          1.4.3     2019-03-12 [1]
#>  stringr          1.4.0     2019-02-10 [1]
#>  testthat         2.2.1     2019-07-25 [1]
#>  tibble           2.1.3     2019-06-06 [1]
#>  units            0.6-5     2019-10-08 [1]
#>  usethis          1.5.1     2019-07-04 [1]
#>  V8               2.3       2019-07-02 [1]
#>  withr            2.1.2     2018-03-15 [1]
#>  xfun             0.10      2019-10-01 [1]
#>  XML              3.98-1.20 2019-06-06 [1]
#>  xtable           1.8-4     2019-04-21 [1]
#>  yaml             2.2.0     2018-07-25 [1]
#>  source                               
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  Github (thomasp85/shinyFiles@35d8c90)
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.0)                       
#>  CRAN (R 3.6.1)                       
#>  CRAN (R 3.6.0)                       
#> 
#> [1] C:/Users/FRS/Documents/R/win-library/3.6
#> [2] C:/R/R-3.6.1/library
```

</details>

**Comment by lbusett** [comment]
@Pakillo thanks for checking. If wmic is now (permanently) on your path, running `where wmic` (as done by shinyFiles) on a `cmd` should give you its path. If not, then probably wmic is still not on the path. 

@vnijs Given @pakillo test, simply using `%SystemRoot%\System32\Wbem\wmic` in the system call should work. I could draft a PR if you wish. Would you prefer checking for "bare" wmic first and then eventually for `%SystemRoot%\System32\Wbem\wmic` if it fails,  or just immediately try with `%SystemRoot%\System32\Wbem\wmic` ) 

Lorenzo

**Comment by Pakillo** [comment]
> If wmic is now (permanently) on your path, running where wmic (as done by shinyFiles) on a cmd should give you its path. If not, then probably wmic is still not on the path.

@lbusett you're right. I think running 
```
old_path <- Sys.getenv("PATH")
new_path <- "C:/windows/System32/Wbem"
Sys.setenv(PATH = paste(old_path, new_path, sep = ";"))
```
as I did yesterday does not change the path permanently, but only for that R session. So wmic is indeed not on Windows path, and that's why `where wmic` does not find it.
Then I wonder if calling wmic is necessary, as the shiny app seems to launch correctly, even if wmic is not available.

**Comment by lbusett** [comment]
After the correction on `shinyFiles`, the app will work also if you do not have wmic in the path, BUT I'd guess that you will have some limitations on the selection of the folders where you want to install the required dependencies. The same will probably happen for the selection of folders where processed data will be stored (from the main sen2r GUI).

**Comment by Pakillo** [comment]
Ah I see... thanks for the explanation. Yes, indeed, the shiny app fails as soon as we have to specify directories. So we must make wmic available somehow.
FYI I'm happy to add wmic to the path myself, and I can run everything now, so now I'm just thinking of easing this for future users.
Thanks

**Comment by lbusett** [comment]
Thanks for the testing. I'll propose a PR to ShinyFIles soon, in order to use the full path to wmic and avoid the problem. I may ask you to test if it solves the issue afterwards.

**Comment by Pakillo** [comment]
Sure! Let me know if I can be of any help. Happy to test things. Cheers

**Comment by lbusett** [comment]
@Pakillo Ok, I tried a fix. Could you try installing: 

``` r
remotes::install_github("lbusett/shinyFiles", ref = "fixwmicpath")
```

and try again? 

  thanks, 

Lorenzo

**Comment by vnijs** [comment]
Ah. Seems like to may have been working in parallel @lbusett. 

@Pakillo Can you try the below? Using `Sys.getenv("SystemRoot")` in the code below on Windows seems to do the trick. 

```
wmic <- paste0(Sys.getenv("SystemRoot"), "\\System32\\Wbem\\wmic")
volumes <- system(paste(wmic, "logicaldisk get Caption"), intern = T)
```

You can install from:

`remotes::install_github("thomasp85/shinyFiles")`

**Comment by Pakillo** [comment]
I've just tried with the last version of shinyFiles and it works. No errors, no warnings, able to browse folders.
Thank you @lbusett @vnijs !!

**Comment by lbusett** [comment]
Good, thanks for the testing! I am closing this now, but feel free to reopen in case you encounter other problems.

**Comment by vnijs** [comment]
I added a "fail-safe" in case WMIC.exe is not in the usual place for some reason and created binaries to install. @Pakillo If you wouldn't mind testing one more by installing the below I'd very much appreciate it.

```
install.packages("shinyFiles", repos = "https://radiant-rstats.github.io/minicran/")
```

**Comment by Pakillo** [comment]
No problem! Tested in my Windows 7 - no problem found
Cheers
